static/js/buddy_list.js

```diff
@@ -29,7 +29,7 @@ function buddy_list_conf() {
 
     conf.get_key_from_li = function (opts) {
         const user_id = opts.li.expectOne().attr('data-user-id');
-        return user_id;
+        return parseInt(user_id, 10);
```

Comment:
Just to add this as a diff comment so it doesn’t get forgotten: the `parseInt(elem.attr("data-user-id"), 10)` pattern should be replaced with `elem.data("user-id")` everywhere. (There are many instances of this and I won’t spam them all with comments.)

Reply:
In buddy_list it's nice to use `data-user-id` explicitly, since it's a bit easier to grep for and we also use it to build up selectors like `conf.item_sel + "[data-user-id='" + user_id + "']";`.  It seems like it's more explicit to do `parseInt` ourselves, since we're not expecting arbitrary JSON here.

Reply:
done as separate commit